ccm_web/client/src/pages/AddUMUsers.tsx

```diff
@@ -143,6 +144,9 @@ function AddUMUsers (props: AddUMUsersProps): JSX.Element {
     }
   )
 
+  const apiErrors = [getCanvasSectionDataError]
+  if (hasUnauthorized(apiErrors)) location.href = '/'
```

Comment:
I wish we have some kind of interceptor on 401 in react as well, we are repeating these 2 line of code everywhere.

Reply:
Yeah, it's not ideal. I'd like to just use this once on each page, but not all API calls are occurring at the top level (as in `CreateSectionWidget`).

I guess one alternative would be to send some way for the frontend to know it needs to redirect (like an error property), and then maybe we could do it `handleErrors`. I could try that. I'm less confident about the frontend solution, although I'm not THAT confident about any of this.

Reply:
You may lose some readability though with moving it to a utility function, it's kinda nice to see, okay, if I'm using this component and an unauthorized error occurs, I'm going to get sent back to root.

Reply:
@pushyamig, does this seem better? https://github.com/ssciolla/canvas-course-manager-next/commit/452c3526787100ad2611304392965f5a722a90f4

Reply:
Yup. Looks good! Thx for having this fix at the top level. This way we don't need to worry about  adding a redirect for every API call in future. Since these things can easy to miss.

Reply:
Okay, change is incorporated. Seems better to me overall, much easier to keep things in sync over time. It's a little strange to me to be redirecting from this error handling abstraction layer, but other benefits outweigh that strangeness.